### PR TITLE
Log deletes of all descendent pages during delete

### PR DESCRIPTION
**Pull request by RealOrangeOne**
Currently, when deleting a page, only the page itself and its direct children are logged. This change adds log entries for all descendents, and thus all the pages which are deleted. The crux of the issue being that `.get_children()` only returns direct children.

This will slow down deletes slightly, but make them more correct. To counter this, I've added `.specific` to the queryset.

_Please check the following:_

- [x] Do the tests still pass?[^1]
- [x] Does the code comply with the style guide? 
    - [ ] Run `make lint` from the Wagtail root. 
- [x] For Python changes: Have you added tests to cover the new/fixed behaviour?
- [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    - [ ] **Please list the exact browser and operating system versions you tested**:
    - [ ] **Please list which assistive technologies [^3] you tested**: 
- [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**. 

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Review by zerolab — APPROVED** [review]
Good catch @RealOrangeOne.
LGTM. We should backport to 2.16.x too

File: wagtail/tests/test_audit_log.py [hunk 1]
```diff
@@ -278,28 +278,31 @@ def test_page_delete(self):
...
+        child.add_child(
+            instance=SimplePage(
+                title="Grandchild", slug="grandchild-page", content="hello"
+            )
         )
 
-        # check deleting a parent page logs child deletion
-        self.home_page.delete()
+        # check deleting a parent page logs descendent deletion
+        with self.assertNumQueries(189):
```
Comment (gasman) [inline wagtail/tests/test_audit_log.py:288]: oops, missed this when merging...

We should avoid these sorts of 'magic numbers' in tests, as they're likely to be highly sensitive to unrelated changes to the logic. Will revert...

**Comment by zerolab** [comment]
#8292 does the 2.16.x backport
